include/inviwo/core/algorithm/linearmap.h

```diff
@@ -35,32 +35,38 @@ namespace inviwo::util {
 
 template <typename T>
 constexpr T linearMap(T val, glm::vec<2, T> from, glm::vec<2, T> to) noexcept {
+    static_assert(std::is_floating_point_v<T>);
     return (val - from.x) / (from.y - from.x) * (to.y - to.x) + to.x;
 }
 
 template <typename T, size_t N>
 constexpr glm::vec<N, T> linearMap(glm::vec<N, T> val, glm::vec<2, T> from,
                                    glm::vec<2, T> to) noexcept {
+    static_assert(std::is_floating_point_v<T>);
     return (val - from.x) / (from.y - from.x) * (to.y - to.x) + to.x;
 }
 
 template <typename T>
 constexpr T linearMapToNormalized(T val, glm::vec<2, T> from) noexcept {
+    static_assert(std::is_floating_point_v<T>);
     return (val - from.x) / (from.y - from.x);
 }
 
 template <typename T, size_t N>
 constexpr glm::vec<N, T> linearMapToNormalized(glm::vec<N, T> val, glm::vec<2, T> from) noexcept {
+    static_assert(std::is_floating_point_v<T>);
     return (val - from.x) / (from.y - from.x);
 }
 
 template <typename T>
 constexpr T linearMapFromNormalized(T val, glm::vec<2, T> to) noexcept {
+    static_assert(std::is_floating_point_v<T>);
```

Comment:
This one does not need to be floating point.
Also, will this not break code that utilize data dispatch functionality?
In most cases it won't make sense to perform computations using the type they are stored.
In my opinon it would probably be better to return float/double, but perhaps another PR as this does not really change the existing behaviour.